src/Microsoft.AspNetCore.Razor/Parser/CSharpCodeParser.Statements.cs

```diff
@@ -464,7 +464,10 @@ private void Statement(Block block)
             var type = CurrentSymbol.Type;
             var loc = CurrentLocation;
 
-            var isSingleLineMarkup = type == CSharpSymbolType.Transition && NextIs(CSharpSymbolType.Colon);
+            //both cases @: and @:: are triggered as markup, second colon in second case will be triggered as a plain text
```

Comment:
Our format for comments is typically `// Hello World`. Note the space and the cap letter.

Also, if you could space out the line below: `(NextIs(CSharpSymbolType.Colon) || NextIs(CSharpSymbolType.DoubleColon));`